src/server/src/main/scala/geotrellis/server/wcs/ops/GetCoverage.scala

```diff
@@ -1,7 +1,62 @@
 package geotrellis.server.wcs.ops
 
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.crop._
+import geotrellis.raster.reproject._
+import geotrellis.raster.io.geotiff._
+import geotrellis.server.wcs.WcsRoute
 import geotrellis.server.wcs.params.GetCoverageWCSParams
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.stitch._
+
+import com.typesafe.config.ConfigFactory
+
+import scala.util.Try
 
 object GetCoverage {
-  def build(params: GetCoverageWCSParams): Array[Byte] = ???
+  def build(catalog: WcsRoute.MetadataCatalog, params: GetCoverageWCSParams): Array[Byte] = {
+    def valueReader = Try(ConfigFactory.load().getString("server.catalog")).toOption match {
```

Comment:
You'd want to use the [`S3CollectionLayerReader`](https://github.com/locationtech/geotrellis/blob/master/s3/src/main/scala/geotrellis/spark/io/s3/S3CollectionLayerReader.scala) to parallelize over fetching of the tiles.

Reply:
You'll probably need to write an SPI interface for `CollectionLayerReader`